frontend/src/pages/NewDoctor.js

```diff
@@ -0,0 +1,156 @@
+import {Button, TextField} from '@mui/material';
+import {useState} from 'react';
+import {submitDoctor} from '../service/doctorApiService';
+import styled from 'styled-components/macro';
+
+export default function NewDoctor() {
+	const initialDoctor = {
```

Comment:
Do you really need this? If you don't declare the initial state like this all the key/value pairs will still be added in your handleChange function. Your backend will set any key = null if it doesn't exist in the json that it receives. You could reset your state just like this setNewDoctor({}).
Also the test if(!newDoctor) will always be false when you declare the initial state like this.
I would rather consider writing a logic that checks for certain attributes that you consider absolutely necessary and prevents the submission when these are not set. i.e.: if(!newDoctor.lastName && !newDoctor.specialty) return

Reply:
If I don't define initialDoctor, I get loads of undefined variables, e.g. all the values of the TextFields (value={newDoctor.firstName} etc.). Also the attributes in the if-condition if(!newDoctor.lastName && !newDoctor.specialty) are unknown. Or am I missing something?

Reply:
Ok, I had just seen the conditional check in the submit function and not considered the usage of the doctor's properties in the components further down.
Anyways the check if(!newDoctor) will always return false, since the newDoctor is initially set. Maybe better check for length of the necessary properties in your submit form. I.e.: if(!newDoctor.lastName.length ...)
Also consider giving feedback to the user wehn the submit doesn't get executed because of missing information.
Or, the best solution I think, would be to just deactivate the button that triggers the submit function when the given properties don't fulfill the requirements.